src/main/java/com/epam/esm/model/GiftCertificate.java

```diff
@@ -0,0 +1,169 @@
+package com.epam.esm.model;
+
+import org.hibernate.annotations.Where;
+import org.hibernate.envers.Audited;
+
+import javax.persistence.*;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+@Entity
+@Table(name = "giftcertificates")
+@Audited
+@Where(clause = "Active = true")
+public class GiftCertificate implements BaseModel {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @NotNull(message = "certificate must have name")
```

Comment:
That's what happens. This annotation allows to do validation at the time the user sends a request, if the data (in this case, name) is incorrect (or empty), then the request will not even get into the controller. Therefore, it will not fall on both the service layer and the database layer.